app/assets/javascripts/request_service.js

```diff
@@ -0,0 +1,34 @@
+var RequestService = function() {
```

Comment:
You probably don't need to add these to the prototype. You can just create a singleton with `{}` and add the properties to get the same effect with less memory overhead.